### ISSUE TITLE
-on-error=ask doesn't prompt when using WSL

### ISSUE THREAD

**Issue by adamrushuk** [hashicorp/packer#9170]
#### Overview of the Issue

I'm using `-on-error=ask` so Packer should ask me what to do when there is an error. This works when I use a PowerShell 7 console, but does not work when using WSL on my Windows 10 computer.

The main error I see in the logs is `Error asking for input: no available tty`. I also see this when running in a Jenkins pipeline, so I assume Packer doesn't realise it can prompt in WSL.

#### Reproduction Steps

Run `packer build -on-error=ask -color=false ubuntu-ask-on-error.json` within WSL. Config file: https://github.com/adamrushuk/Packer-Templates/blob/master/azure-test/ubuntu-ask-on-error.json

### Packer version

`Packer v1.5.6`

### Simplified Packer Buildfile

https://github.com/adamrushuk/Packer-Templates/blob/master/azure-test/ubuntu-ask-on-error.json

### Operating system and Environment details

Running via WSL (Ubuntu 18.04.3 LTS) on Windows 10.

### Log Fragments and crash.log files

`2020/05/05 17:10:30 packer-builder-azure-arm plugin: Error asking for input: no available tty`

**Comment by benhillis** [comment]
How is packer getting user input?  WSL2 doesn't use a tty device, we use pseudoterminals (pts) since we are relaying data to and from the host console.

**Comment by adamrushuk** [comment]
@benhillis are you saying this cannot be done as WSL can't prompt for user input?

**Comment by benhillis** [comment]
No, I'm saying that we don't use tty devices, we use pseudo-terminals (for user input).

**Comment by adamrushuk** [comment]
ah ok, thanks for clarifying. Hopefully there is a solution using pseudo-terminals.

**Comment by son-la** [comment]
I have been using v.1.4.3 on WSL1 and -debug flag has worked perfectly. Now when updating to 1.4.4, it stopped working and even the latest 1.6.4 doesn't work.

**Comment by gcupic** [comment]
I can also confirm this unexpected behaviour with packer --version 1.6.5 and Ubuntu 20.04 on WSL...

**Comment by timblaktu** [comment]
I suspect this is not a WSL issue. I am also experiencing an ineffectual `-on-error=ask` when running `packer build` in a Jenkins pipeline. The build just aborts on error, immediately. Like the OP I am trying to get packer build to pause so I can investigate before the target machine is taken down. I assume the problem is that packer requires a real TTY in order to execute the pause.

I have found a useful workaround to instead use `-on-error=abort` which prevents the subsequent cleanup steps from running, which leaves the target machine in existence. In my case this is means I can remote into the remote machine on Jenkins or vSphere after the error..

It'd be MUCH BETTER, though, for this issue to be fixed, for `-on-error=ask` to work even when there's no TTY, so that the cleanup steps still automatically run after my "debug session" is over. As it is now, I have to (well, in my case _Jenkins_ has to) manually do this every time. 

I would propose a new `-on-error=pause` setting(s) that would prescribe the behavior through configuration by adding support for `-on-error=pause` and a new `packer build` arg named `-on-error-pause-time=<time-string>`, e.g. `-on-error-pause-time=10m`.

**Comment by adamrushuk** [comment]
I agree that `-on-error=pause` would be a good idea.

I forgot to follow up on this, as I came up with a similar solution and blogged about it here: https://adamrushuk.github.io/interactive-debugging-within-a-jenkins-container-running-on-kubernetes/

Using `error-cleanup-provisioner` can be used to "pause on error".

**Comment by timblaktu** [comment]
Thanks for that, in my case I cannot use the error-cleanup-provisioner bc I'm using HCL and [that construct is not yet supported](https://github.com/hashicorp/packer/issues/9042). I will just write a shell wrapper around my packer call to catch non-zero return codes and sleep..

**Comment by riebecj** [comment]
We're over a year later and I'm having this issue on Mac OS attempting to build both Linux and Windows AWS AMIs. The breakpoint provisioner and the `-on-error=ask` flag both give the same error of `Error asking for input: no available tty`
